### PR TITLE
Add database driver for SQLCipher

### PR DESCRIPTION
**Pull request by klingtnet**
This is a copy of the existing sqlite3 driver, except that it imports `_
"github.com/mutecomm/go-sqlcipher/v4"` as database driver.  This allows to use migrate with encrypted sqlite databases.

### CODE REVIEW

**Comment by klingtnet** [comment]
Here's a small demo application that I used against my local fork:

```go
package main

import (
        "database/sql"
        "errors"
        "fmt"
        "log"
        "net/url"

        "github.com/golang-migrate/migrate/database/sqlcipher"
        "github.com/golang-migrate/migrate/v4"
        _ "github.com/golang-migrate/migrate/v4/source/file"
        _ "github.com/mutecomm/go-sqlcipher/v4"
)

func main() {
        key := url.QueryEscape("secret")
        dbname := fmt.Sprintf("file:example.db?_pragma_key=%s&_pragma_cipher_page_size=4096", key)
        db, err := sql.Open("sqlite3", dbname)
        if err != nil {
                log.Fatal("sql.Open: ", err)
        }
        driver, err := sqlcipher.WithInstance(db, &sqlcipher.Config{})
        if err != nil {
                log.Fatal("sqlcipher.WithInstance: ", err)
        }

        m, err := migrate.NewWithDatabaseInstance(
                "file://migrations",
                "sqlcipher", driver)
        if err != nil {
                log.Fatal("migrate.NewWithDatabaseInstance: ", err)
        }
        err = m.Up()
        if err != nil {
                if errors.Is(err, migrate.ErrNoChange) {
                        log.Println("already migrated, nothing to do")
                } else {
                        log.Fatal(err)
                }
        }
}
```

**Comment by klingtnet** [comment]
> ## Pull Request Test Coverage Report for [Build 851](https://coveralls.io/builds/33183041)
> 
>     * **118** of **176**   **(67.05%)**  changed or added relevant lines in **1** file are covered.
> 
>     * No unchanged relevant lines lost coverage.
> 
>     * Overall coverage increased (+**0.5%**) to **53.733%**
> 
> 
> Changes Missing Coverage 	Covered Lines 	Changed/Added Lines 	%
> [database/sqlcipher/sqlcipher.go](https://coveralls.io/builds/33183041/source?filename=database%2Fsqlcipher%2Fsqlcipher.go#L44) 	118 	176 	67.05%
> Totals 	[Coverage Status](https://coveralls.io/builds/33183041)
> Change from base [Build 846](https://coveralls.io/builds/33093942): 	0.5%
> Covered Lines: 	2764
> Relevant Lines: 	5144
> ##### yellow_heart - [Coveralls](https://coveralls.io)

I could not find a failing even though the CI run with Go 1.14 exited with `1`.

File: database/sqlcipher/README.md [hunk 1]
```diff
@@ -0,0 +1,3 @@
+# sqlcipher
+
+This is just a copy of the sqlite3 driver except that it imports `github.com/mutecomm/go-sqlcipher`.
```
Comment (dhui) [inline database/sqlcipher/README.md:3]: link to the [sqlite3](https://github.com/golang-migrate/migrate/tree/master/database/sqlite3) db driver

Comment (klingtnet) [reply database/sqlcipher/README.md]: Everything done.

**Review by dhui — APPROVED** [review]
Waiting for tests to pass

**Pull request merged**